src/main/java/com/google/sps/ServerMain.java

```diff
@@ -8,13 +8,49 @@
 import org.eclipse.jetty.webapp.Configuration;
 import org.eclipse.jetty.webapp.WebAppContext;
 import org.eclipse.jetty.webapp.WebInfConfiguration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.time.Instant;
+import java.time.Duration;
+import java.util.Date;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.Timestamp;
 
 /**
  * Starts up the server, including a DefaultServlet that handles static files, and any servlet
  * classes annotated with the @WebServlet annotation.
  */
 public class ServerMain {
 
+  private static void deleteOldEvents() {
```

Comment:
Can you put all of this deletion / threading logic in a different class?? Maybe something like `DatastoreGarbageCollector` and you can have a `start` method and a `stop` method which contains the logic that you have below?

Reply:
This has been done. It definitely looks better now!